beacon_node/http_api/src/block_id.rs

```diff
@@ -266,6 +267,29 @@ impl BlockId {
             Err(e) => Err(warp_utils::reject::beacon_chain_error(e)),
         }
     }
+
+    pub async fn blob_sidecar_list_indexed<T: BeaconChainTypes>(
+        &self,
+        indices: BlobIndicesQuery,
+        chain: &BeaconChain<T> 
+    ) -> Result<BlobSidecarList<T::EthSpec>, warp::Rejection> {
+        let blob_sidecar_list = self.blob_sidecar_list(&chain).await?;
+        let blob_sidecar_list_indexed = match indices.indices {
+            Some(vec) => {
+                let list = blob_sidecar_list.into_iter()
+                .enumerate()
+                .filter(|(index, _)| vec.contains(index))
```

Comment:
I think it would be more accurate to filter using the `blob_sidecar.index` instead, in case if it's somehow not the same order in the list. This means we can also get rid of `.enumerate`.